### PR TITLE
Cluster deployment documentation refactor, adds advanced instructions

### PR DESCRIPTION
**Pull request by jmspring**

### CODE REVIEW

File: cluster/environments/azure-advanced/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+*NOTE*: Because the deployment of Flex Volume requires additional [Azure Role Assignments](https://docs.microsoft.com/en-us/rest/api/authorization/roleassignments) be created.  The Service Principal provisioned as part of the [common setup](../../azure/README.md#create-an-azure-service-principal) is required to have `Owner` level privileges on the subscription.
+
+### Azure Key Vault
+
+[Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/) is a secure repository for secets, keys, and certificates.  Within this environment, it is used to store and manage secrets that could be used by the AKS deployment itself or applications running within the cluster.  In order to configure Azure Key Vault in this environment, the following additional variables must be definted:
+
+- `keyvault_name`: this is a unique name for the Key Vault.  Per [this doc](https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates), Key Vault names are are globally unique and must be a 3-24 character string, containing only 0-9, a-z, A-Z, and -.
+- `secret_name`: the name of a secret to create within the Key Vault
```
Comment (timfpark) [inline cluster/environments/azure-advanced/README.md]: Can we add these to the terraform.tfvars in the template so folks know they need to fill them out?  Also, let's add some description of what these do - its not clear to me from the description why we need to create secrets here?

File: cluster/azure/README.md [hunk 2]
```diff
@@ -16,44 +16,201 @@ Additionally, you need the Azure `az` command line tool in order to create and f
...
+
+```bash
+>  az storage account keys list --account-name <storage account name>
+```
+
+With this, update `backend.tfvars` file in your cluster environment directory with these values and use `terraform init -backend-config=./backend.tfvars` to setup usage of the Azure backend.
+
+### Configure `kubectl` to see your new AKS cluster
+
+Upon deployment of the cluster, one artificat that the `terraform` scripts generate is the credentials necessaryfor logging into the AKS cluster that was deployed.  These credentials, are placed in the location specified by the variable `output_directory`.  For single cluster environments, this defaults to `./output`.  For multi cluster environments, the default `output_directory` will be documented in those environments.
```
Comment (timfpark) [inline cluster/azure/README.md]: typo in artifact and "necessaryfor"

File: cluster/environments/azure-advanced/README.md (hunk 1 above)
Comment (jmspring) [reply cluster/environments/azure-advanced/README.md]: I'll update this, is it needed?  Unknown.  But showing how to do such is useful.

**Comment by jmspring** [comment]
Closes #191

**Pull request merged**